.github/workflows/create-release.yml

```diff
@@ -49,13 +42,18 @@ jobs:
           file: "v2/out/release/*"
           file_glob: true
 
-      - name: Login to registry
-        # note that all creds are on host and never passed into devcontainer
-        uses: docker/login-action@v2.1.0
-        with:
-          registry: ${{ secrets.REGISTRY_LOGIN }}
-          username: ${{ secrets.AZURE_CLIENT_ID }}
-          password: ${{ secrets.AZURE_CLIENT_SECRET }}
+      - name: Docker login
+        run: |
+          container_id=${{env.container_id}}
+          docker exec -e AZURE_CLIENT_ID -e AZURE_CLIENT_SECRET -e DOCKER_REGISTRY "$container_id" task docker-login
+        env:
+          DOCKER_REGISTRY: ${{ secrets.REGISTRY_LOGIN }}
+          AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
 
-      - name: Push docker image
-        run: docker push --all-tags ${{ secrets.REGISTRY_PUBLIC }}/azureserviceoperator
+      - name: Build, tag and push docker image
+        run: |
+          container_id=${{env.container_id}}
+          docker exec -e DOCKER_PUSH_TARGET "$container_id" task controller:docker-push-multiarch
```

Comment:
We don't have any other docker push target instead of `docker-push-local`. I'm letting the local push target be there for reasons: 
- For local builds, we don't need multi-arch images as they take time to build and occupy space.
- Buildx does not have direct support for pushing to local registries. We need to add in a few extra steps to enable it. Mentioned in [Build cannot export to registry on localhost](https://github.com/docker/buildx/issues/94#top)

Reply:
We should probably document this on the taskfile targets, for why we have two and why different ones are used in different places.

There are some risks with this too (we're not actually testing the image we're creating), but I see that _lots_ of people have this problem, as documented:
https://github.com/docker/buildx/issues/166
https://github.com/docker/buildx/issues/1152

There _does_ seem to be a way to push buildx image to local registry now though, so I am wondering if we should try swapping the `kind` steps to use the multiarch image?

Reply:
We can surely swap kind steps to use buildx, but not sure about the multiarch images if thats a good idea. As I noticed, for cross-platform images buildx takes around about 10-15 mins to build. Which would increase the CI run time.

Reply:
CI shouldn't be building the image every time - it's supposed to be cached. We even have [a workflow](https://github.com/Azure/azure-service-operator/blob/main/.github/workflows/build-devcontainer-image.yml )that runs weekly to keep it current.

So if switching CI to use buildx slows it down, we need to fix that.

Reply:
I don't think CI caches this image @theunrepentantgeek - it caches the .devcontainer image, not the controller image (being built here). It must build the controller image every time because that's the image that:
- Contains the code changes for the PR in question.
- Runs in kind and is verified by the tests.

My understanding of what `buildx` does is that it basically calls N docker builds with different architectures and then automatically merges the manifests. So it may be OK to not use it for the local build (see the issues I linked above for where this is talked about some and people suggest using buildx w/ a single version argument to get local working, which anyway is going to be different than what we do for release)